src/block-data-providers/bitcoin-core/provider.ts

```diff
@@ -0,0 +1,278 @@
+            };
+            return await this.request(body);
+        } catch (error) {
+            this.logger.log(
+                `Error fetching transaction with transaction id : ${txid}`,
+            );
+            throw error;
+        }
+    }
+
+    public async processBlock(height: number): Promise<Transaction[]> {
+        const parsedTransactionList: Transaction[] = [];
+        const blockHash = await this.getBlockHash(height);
+        this.logger.log(
+            `Processing block at height ${height}, hash ${blockHash}`,
+        );
+        const block = await this.getBlock(blockHash, 2);
+        for (const txn of block.tx) {
+            const parsedTransaction = await this.parseTransaction(
+                txn,
+                block.hash,
+                block.height,
+            );
+            parsedTransactionList.push(parsedTransaction);
+        }
+        return parsedTransactionList;
+    }
+
+    private async parseTransaction(
+        txn: BlockTransaction,
+        blockHash: string,
+        blockHeight: number,
+    ): Promise<Transaction> {
+        const inputs: TransactionInput[] = await Promise.all(
+            txn.vin.map(
+                async (input) => await this.parseTransactionInput(input),
+            ),
+        );
+        const outputs: TransactionOutput[] = txn.vout.map((output) =>
+            this.parseTransactionOutput(output),
+        );
+
+        return {
+            txid: txn.txid,
+            vin: inputs,
+            vout: outputs,
+            blockHeight,
+            blockHash,
+        };
+    }
+
+    private async parseTransactionInput(
+        txnInput: Input,
+    ): Promise<TransactionInput> {
+        let txid =
+            '0000000000000000000000000000000000000000000000000000000000000000';
+        let vout = 0;
+        let prevOutScript = '';
+        let witness = undefined;
+        let scriptSig = '';
+
+        if (!isCoinbaseInput(txnInput)) {
+            txid = txnInput.txid;
+            const prevTransaction = await this.getRawTransaction(
+                txnInput.txid,
+                true,
+            );
+            vout = txnInput.vout;
+            prevOutScript = prevTransaction.vout.find((out) => out.n == vout)
+                .scriptPubKey.hex;
+            witness = txnInput.txinwitness;
+            scriptSig = txnInput.scriptSig.hex;
+        }
+
+        return {
+            txid,
+            vout,
+            scriptSig,
+            witness,
+            prevOutScript,
+        };
+    }
+
+    private parseTransactionOutput(txnOutput: Output): TransactionOutput {
+        return {
+            scriptPubKey: txnOutput.scriptPubKey.hex,
+            value: txnOutput.value,
+        };
+    }
+
+    async request(body: RPCRequestBody): Promise<any> {
+        const { rpcUser, rpcPass, rpcPort, rpcHost } = this.config;
+        try {
+            const response = await axios.post(
+                `http://${rpcHost}:${rpcPort}/`,
```

Comment:
add a private read-only property called url or something and set it inside the constructor.

Reply:
I don't think it is necessary, since it is only used in a single function, i have localised such utility to the particular function.

Reply:
even if the utility is localized, we are creating an additional string that is instantiated multiple times even though its value is not changing. Since `request` can be called a lot, I feel what I suggested is an optimisation.

Reply:
@aruokhai I think you missed this